exporter/exporter.go

```diff
@@ -153,12 +156,23 @@ func (e *Exporter) scrape(groupScrapes chan<- GroupScrapeResult) {
 	var errorCount uint64 = 0
 
 	asgSvc := autoscaling.New(e.session, aws.NewConfig())
+
 	err := asgSvc.DescribeAutoScalingGroupsPages(&autoscaling.DescribeAutoScalingGroupsInput{
 		AutoScalingGroupNames: aws.StringSlice(e.groups),
 	}, func(result *autoscaling.DescribeAutoScalingGroupsOutput, lastPage bool) bool {
 		log.Debugf("Number of AutoScaling Groups found: %d [lastPage = %t]", len(result.AutoScalingGroups), lastPage)
 		var wg sync.WaitGroup
 		for _, asg := range result.AutoScalingGroups {
+			if e.groupsFilter != "" && asg.AutoScalingGroupName != nil {
+				matched, err := regexp.MatchString(e.groupsFilter, *asg.AutoScalingGroupName)
+				if err != nil {
+					log.WithError(err).Error("Regex matching failed")
+					atomic.AddUint64(&errorCount, 1)
+				}
+				if !matched {
```

Comment:
One minor comment (without testing it): I assume if `err` is non-`nil` that `matched` will always be `false`, i.e. an incorrect regex will cause no ASGs to be exported, ever. Part of me feels like this is good behaviour, and another part feels like it should then just try to export everything.

In the context of this PR I think we should just leave it like this, I just didn't want to skip mentioning it.

Reply:
I would think, that returning no ASGs is more alarming, then exporting all ASGs. So if one provides a regex that is broken, it is better to really return nothing.